File: app/src/main/java/com/nicholasdoglio/eyebleach/di/AppComponent.java
/*
    Aww Gallery
    Copyright (C) 2017  Nicholas Doglio

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.nicholasdoglio.eyebleach.di;

import android.app.Application;

import com.nicholasdoglio.eyebleach.AwwGalleryApp;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;

/**
 * @author Nicholas Doglio
 */
@Singleton
@Component(modules = {
        AndroidInjectionModule.class,
        AppModule.class,
        NetworkModule.class,
        DatabaseModule.class,
        ActivityBindingModule.class})
public interface AppComponent {

    void inject(AwwGalleryApp application);

    @Component.Builder
    interface Builder {

        @BindsInstance
        Builder application(Application application);

        AppComponent build();
    }
}

File: mobile/src/main/java/com/nicholasdoglio/eyebleach/di/AppComponent.kt
package com.nicholasdoglio.eyebleach.di

import android.app.Application
import com.nicholasdoglio.eyebleach.AwwGalleryApp
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

/**
 * @author Nicholas Doglio
 */
@Singleton
@Component(
  modules = [
    (AndroidSupportInjectionModule::class),
    (AppModule::class),
    (MainActivityBindingModule::class)]
)
interface AppComponent : AndroidInjector<AwwGalleryApp> {

  @Component.Builder
  interface Builder {
    @BindsInstance
    fun application(application: Application): Builder

    fun build(): AppComponent
  }
}